packages/backend/src/services/UserService.ts

```diff
@@ -488,6 +488,35 @@ export class UserService extends BaseService {
         }
     }
 
+    /**
+     * In this method we check if the user is allowed to register in this instance before registerUser happens.
+     * If the instance is a single org, and there are already users and orgs,
+     * and the user's email domain is allowed in the org whitelisted,
+     * we throw a ForbiddenError.
+     * @param inviteCode invite code
+     * @param email email for the user registering
+     */
+    async checkNewUserRegistrationAllowed(
+        inviteCode: string | undefined,
+        email: string,
+    ) {
+        if (
+            inviteCode === undefined &&
+            !this.lightdashConfig.allowMultiOrgs &&
+            (await this.userModel.hasUsers()) &&
+            (await this.organizationModel.hasOrgs()) &&
+            (
+                await this.organizationModel.getAllowedOrgsForDomain(
+                    getEmailDomain(email),
+                )
+            ).length !== 1
```

Comment:
Better check if bellow 1 in case `allowMultiOrgs` is disabled after a couple of orgs are created. 
```suggestion
            ).length < 1
```

Reply:
This check is for `allowedOrgsForAdmin` where we should only return 1 org if the email is whitelisted by an org. 

In here we are checking `===1` , because we only should have 1 matching org https://github.com/lightdash/lightdash/blob/main/packages/backend/src/services/UserService.ts#L695. Do you think we  should we change that logic too ?

Reply:
> This check is for allowedOrgsForAdmin where we should only return 1 org if the email is whitelisted by an org.

You meant `getAllowedOrgsForDomain` ? It can return multiple orgs. 

> In here we are checking ===1 , because we only should have 1 matching org

That assumption is incorrect. 
If the user only has 1 matching org, we auto join them to the org.
If the user has more than 1, they will have the option in the UI to decide which one to join.
If there is no matching org, one will be created (logic for this is in FE 😢 ) or throw an error when they can't (flow we are trying to fix).

Having `< 1` would prevent the third flow.